### PR TITLE
Add parsed nix config

### PR DESCRIPTION
**Pull request by Eken-beep**
Add a way to write the gBar configuration in the nix language, like many other home manager modules, this spares, atleast nix users, from the trouble of using a new config language and integrates it nicer into existing home manager configs. Currently not all of gBars options have a nix option and some like SNIIconSize probably wont due to how they're made in the config, I could do it in a single string with both number and name but that looks ugly and there's still extraConfig. I have as ambition to update this while new options are added.

### CODE REVIEW

File: module.nix [hunk 1]
```diff
@@ -5,49 +5,191 @@ self:
...
       '';
     };
 
-    extraCSS = lib.mkOption {
-      type = lib.types.nullOr lib.types.lines;
+    extraCSS = mkOption {
+      type = types.nullOr types.lines;
       default = null;
       description = ''
         Configuration to write to ~/.config/gBar/config, if none nothing happens
```
Comment (scorpion-26) [inline module.nix:172]: Shoudn't this be ~/.config/gBar/style.css?
It's probably something that slipped through the last PR.

File: module.nix [hunk 2]
```diff
@@ -5,49 +5,191 @@ self:
...
+            options = {
+                CPUThermalZone = mkOption {
+                    type = types.nullOr types.str;
+                    default = "/sys/devices/pci0000:00/0000:00:18.3/hwmon/hwmon2/temp1_input";
+                    description = "path to the cpu thermal sensor, probably something in /sys/device";
+                };
+                SuspendCommand = mkOption {
+                    type = types.str;
+                    default = "systemctl suspend";
+                    description = "";
```
Comment (scorpion-26) [inline module.nix:44]: Can you give each option a small description (Like CPUThermalZone). It probably gives a bit more context and you can copy it from the default config.

**Comment by scorpion-26** [comment]
Thanks again for the PR!

> Currently not all of gBars options have a nix option and some like SNIIconSize probably wont due to how they're made in the config, I could do it in a single string with both number and name but that looks ugly and there's still extraConfig. 

I think that should be documented somewhere, that there some other options which can be set via extraConfig.

> I have as ambition to update this while new options are added.

Should I ping you, when there's something new?

File: module.nix [hunk 3]
```diff
@@ -5,49 +5,191 @@ self:
...
+                    default = "T";
+                    description = "";
+                };
+                CenterTime = mkOption {
+                    type = types.bool;
+                    default = true;
+                    description = "";
+                };
+                TimeSpace = mkOption {
+                    type = types.nullOr types.str;
```
Comment (scorpion-26) [inline module.nix:92]: TimeSpace is an int

File: module.nix (hunk 1 above)
Comment (Eken-beep) [reply module.nix:172]: That one's fine, I just chose to not put it under the config submodule, functions at the bottom in config =

File: module.nix (hunk 2 above)
Comment (Eken-beep) [reply module.nix:44]: yep but, also should all options have set default values or should they just be empty?

**Comment by Eken-beep** [comment]
> Should I ping you, when there's something new?

Well there's two solutions to that. First I could probably update this file when new features arrive and thats fine. We could also do it like, for example hyprland, and have a solution that just writes whatever into the config file and doesn't do any typechecks. It's easier but IMO a worse solution.

File: module.nix (hunk 2 above)
Comment (scorpion-26) [reply module.nix:44]: I think so. Even though gBar has hardcoded defaults for most configs and doesn't need them, setting them in module.nix too could make things easier to understand.

**Comment by scorpion-26** [comment]
The hyprland one sounds like a bad solution. It doesn't check types, which I doubt gBar likes when there's a type mismatch. It shouldn't be too much work to just update module.nix (If you don't want to do that, I can do it as well though)

**Comment by Eken-beep** [comment]
That should be everything, should be clear how it works now LMK if you think some option is wrong. I can definitely update it however should just be as simple as adding a new option that looks like the previous ones as long as it uses a primitive data type (unlike sni config)

**Comment by scorpion-26** [comment]
Just noticed, ```CheckPackagesCommand``` and ```CheckUpdateInterval``` are missing. Other than that, lgtm.

**Comment by Eken-beep** [comment]
I feel like those are basically redundant as you don't really check for that on nixos (as far as i know)

**Comment by scorpion-26** [comment]
Makes sense. 
Thanks for your effort!

**Pull request merged**